src/Command/CheckAllSchemaTemplatesDefaultTypeCommand.php

```diff
@@ -0,0 +1,200 @@
+        $directory = $input->getArgument('schemaTemplateDirectory');
+        $avroFiles = SchemaFileHelper::getAvroFiles($directory);
+
+        $io = new SymfonyStyle($input, $output);
+
+        $failed = [];
+
+        if (false === $this->checkSchemas($avroFiles, $failed)) {
+            $io->error('Following schema templates have invalid default value types:');
+            $io->listing($failed);
+
+            return 1;
+        }
+
+        $io->success('All schema templates have valid default value types');
+
+        return 0;
+    }
+
+    /**
+     * @param array<string, mixed> $avroFiles
+     * @param array<string, mixed> $failed
+     * @return boolean
+     */
+    private function checkSchemas(array $avroFiles, array &$failed = []): bool
+    {
+        $failed = [];
+
+        foreach ($avroFiles as $schemaName => $avroFile) {
+
+            /** @var string $localSchema */
+            $localSchema = file_get_contents($avroFile);
+
+            $invalidFields = $this->checkDefaultType($localSchema);
+
+            if (count($invalidFields)) {
+                foreach ($invalidFields as $invalidField) {
+                    $failed[] = $invalidField;
+                }
+            }
+        }
+
+        return 0 === count($failed);
+    }
+
+    /**
+     * @param string $localSchema
+     * @return array<string, mixed>
+     */
+    private function checkDefaultType(string $localSchema): array
+    {
+        $decodedSchema = json_decode($localSchema);
+        if (!property_exists($decodedSchema, 'fields')) {
+            return [];
+        }
+
+        $result = [
+            'found' => [],
+            'default' => [],
+        ];
+
+        $result = $this->checkAllFields($decodedSchema, $result);
+
+        return array_diff($result['default'], $result['found']);
+    }
+
+    /**
+     * @param mixed $decodedSchema
+     * @param array<string, mixed> $result
+     * @return array<string, mixed>
+     */
+    private function checkAllFields($decodedSchema, array $result): array
+    {
+        foreach ($decodedSchema->fields as $field) {
+            $fieldTypes = $field->type;
+            if (property_exists($field, 'default')) {
+                $result['default'][] = $this->getFieldName($decodedSchema, $field);
+            }
+
+            if (!is_array($fieldTypes)) {
+                $fieldTypes = [$fieldTypes];
+            }
+
+            foreach ($fieldTypes as $fieldType) {
+                $result = $this->checkSingleField($fieldType, $field, $decodedSchema, $result);
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param mixed $fieldType
+     * @param mixed $field
+     * @param mixed $decodedSchema
+     * @param array<string, mixed> $result
+     * @return array<string, mixed>
+     */
+    private function checkSingleField($fieldType, $field, $decodedSchema, array $result): array
+    {
+        $defaultType = null;
+
+        if (property_exists($field, 'default')) {
```

Comment:
obsolete, we already know that it must exits, we could actually also pass it from before :v: